planner/languages/java/java_planner.go

```diff
@@ -36,38 +52,42 @@ func (p *Planner) Name() string {
 }
 
 func (p *Planner) IsRelevant(srcDir string) bool {
-	// Checking for pom.xml (maven) only for now
-	// TODO: add build.gradle file detection
-	pomXMLPath := filepath.Join(srcDir, "pom.xml")
-	return plansdk.FileExists(pomXMLPath)
+	pomXMLPath := filepath.Join(srcDir, mavenFileName)
+	buildGradlePath := filepath.Join(srcDir, gradleFileName)
+	return plansdk.FileExists(pomXMLPath) || plansdk.FileExists(buildGradlePath)
 }
 
 func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
 	// Creating an empty plan so that we can communicate an error to the user
 	plan := &plansdk.Plan{
-		DevPackages: []string{
-			defaultMaven,
-		},
+		DevPackages: []string{},
 	}
-	javaPkg, err := getJavaPackage(srcDir)
+
+	pomXMLPath := filepath.Join(srcDir, mavenFileName)
+	buildGradlePath := filepath.Join(srcDir, gradleFileName)
+	var builderTool string
+	if plansdk.FileExists(pomXMLPath) {
+		builderTool = MavenType
+	} else if plansdk.FileExists(buildGradlePath) {
+		builderTool = GradleType
+	} else {
+		err := errors.New("Could not locate a Maven or Gradle file.")
+		return plan.WithError(err)
+	}
+	devPackages, err := p.devPackages(srcDir, builderTool)
 	if err != nil {
 		return plan.WithError(err)
 	}
-	startCommand, err := p.startCommand(srcDir)
+	runtimePackages := p.runtimePackages(builderTool)
+	startCommand, err := p.startCommand(srcDir, builderTool)
```

Comment:
Does this `p.startCommand()` get called when user has an override?
I added the error because I'm parsing an xml file looking for specific fields and if there is an error in that parsing, I need to communicate that to the user somehow.

Reply:
Yes, it does. So the planner is deterministic regardless of what the user overrides, but the user override will take precedent.